pkg/health/checker/checker.go

```diff
@@ -112,6 +113,7 @@ func (c consulHealthChecker) WatchHealth(
 				}
 			} else {
 				select {
+				case <-resultCh: // we drain this channel prior to attempting a write so that the freshest value is always available
```

Comment:
you want this to be in a separate `select` because if this case is ready then you won't write healthResults at all